### PR TITLE
Added edit links to project search results

### PR DESCRIPTION
**Pull request by ee-usgs**
Edit links are added when the project has a contact with the same email address of the current user.
Currently the edit link doesn't do anything - Need integration to tie to the edit form.

### CODE REVIEW

File: glri-catalog-ui/src/main/webapp/js/app/sciencebase.js [hunk 1]
```diff
@@ -239,6 +239,29 @@ function($http, Status, FocusAreaManager, $rootScope){
...
+		//Add edit permissions to projects if the user is logged in
+		if (item.resource == 'project' && userService.getUserName() != null) {
+			var userName = userService.getUserName();
+			
+			if ( isDefined(item.contacts) ) {
+				for (var j = 0; j < item.contacts.length; j++) {
+					var contact = item.contacts[j];
+
+					if ( isDefined(contact.email) ) {
+						if (contact.email == userName) {
```
Comment (ee-usgs) [inline glri-catalog-ui/src/main/webapp/js/app/sciencebase.js]: I think it is safe to say that any contact email listed on the project is OK to allow to edit.  I’ll pose that question on the JIRA task.

File: glri-catalog-ui/src/main/webapp/templates/contentFocusArea.html [hunk 2]
```diff
@@ -32,6 +32,9 @@ <h1 class="focusAreaTitle">{{currentFocusArea().name}} Projects</h1>
 			<li ng-repeat="project in status.currentProjectList | orderBy:'dateCreated':true" ng-click="selectProject(project.item)">
 				
 				<h3 ng-bind-html="project.title"></h3>
+				<div class="edit-link" ng-if="project.userCanEdit">
```
Comment (ee-usgs) [inline glri-catalog-ui/src/main/webapp/templates/contentFocusArea.html]: This is currently not working.  I can see that my code is called to add the userCanEdit attribute to project records, but for some reason, the records do not end up w/ the userCanEdit assigned.  I not sure what exactly is happening, but I think it is getting at the weirdness of the Status object and what it means to have a currentProjectList.

File: glri-catalog-ui/src/main/webapp/templates/project-record.html [hunk 3]
```diff
@@ -4,6 +4,9 @@
 			<img ng-src="style/image/darkblue/{{record.resource}}.svg" />
 		</a>
 	</div>
+	<div class="edit-link" ng-if="record.userCanEdit">
+		<a class="btn btn-small" title="Because you have a role on this project, you may edit it.  Click to edit" href="{{record.url}}" target="_blank">Edit this project</a>
```
Comment (ee-usgs) [inline glri-catalog-ui/src/main/webapp/templates/project-record.html]: This works, but we need to update the href to do whatever it is that it needs to do to open this Project in the edit form.

File: glri-catalog-ui/src/main/webapp/templates/contentFocusArea.html (hunk 2 above)
Comment (thongsav-usgs) [inline glri-catalog-ui/src/main/webapp/templates/contentFocusArea.html]: I can take a look but I wonder if it's a timing issue. Angular is set to watch the list, not items in the list, and it looks like userCanEdit is set during a "processItem" call, which happens after the items are already in the array. If that's true, I'm guessing the items are being updated after this directive has already rendered.

File: glri-catalog-ui/src/main/webapp/templates/project-record.html (hunk 3 above)
Comment (thongsav-usgs) [inline glri-catalog-ui/src/main/webapp/templates/project-record.html]: The existing pattern is to use ng-click to call a controller function. The controller function will can set the new path. The path should be "#/Projects/{item_id}"

File: glri-catalog-ui/src/main/webapp/templates/contentFocusArea.html (hunk 2 above)
Comment (thongsav-usgs) [inline glri-catalog-ui/src/main/webapp/templates/contentFocusArea.html]: It's hackey, but you can force it to update by temporarily setting currentProjectList to empty array and reassigning the list back. Might want to try that at least as a test.

File: glri-catalog-ui/src/main/webapp/js/app/newProjectController.js [hunk 4]
```diff
@@ -289,22 +289,23 @@ function($scope, $http, $filter, $location, Status, ScienceBase, projectsService
...
+	}
+	
+	//check to see if we have a project ID, if so, load/bind the project data and set this form to edit mode
+	//Splits are [empty]/Projects/[ProjectId]
+	var parts = $location.path().split(/\/+/);
+	if(parts.length > 2 && parts[2]) {
+		$scope.editMode = true;
+		var id = parts[1];
+		loadAndBindProject(id);
+	}
```
Comment (ee-usgs) [inline glri-catalog-ui/src/main/webapp/js/app/newProjectController.js]: I uncommented this and fixed a few small errors:
* The url parts are actually 3:  The first is always empty
* length was ‘lenght’ in a few places (a mistake I make in Javascript all the time.  Stupid untyped languages.

This gets us close, but there is a fundamental issues w/ the Status object:  It doesn’t really seem to hold the current list of projects as we would expect.  The filter comes back having found zero records b/c there are zero records in the currentProjectList (when coming from search).

File: glri-catalog-ui/src/main/webapp/templates/project-record.html [hunk 5]
```diff
@@ -4,6 +4,9 @@
 			<img ng-src="style/image/darkblue/{{record.resource}}.svg" />
 		</a>
 	</div>
+	<div class="edit-link" ng-if="record.userCanEdit">
+		<a class="btn btn-small" title="Because you have a role on this project, you may edit it.  Click to edit" prevent-default href="javascript:void(0)" ng-click="nav.setPath('Projects/' + record.id)">Edit this Project</a>
+	</div>
```
Comment (ee-usgs) [inline glri-catalog-ui/src/main/webapp/templates/project-record.html]: This linkage to the new/edit form now works on the Search page.

**Comment by ee-usgs** [comment]
I have some hanging concurrent changes, so I’m merging this before its completely ready to go.

**Pull request merged**